Docs Update (#196)
warden-protocol/wardenprotocol: docs/docs/learn/faq.md

# FAQ

## Intents

**Q: What are intents?** 
A: Intents are a set of user-supplied conditions under which a keychain signs a transaction with a private key. They define the criteria that must be met for a transaction to be considered valid and added to the blockchain.

**Q: How do intents work?**

> A: Warden Protocol has an immutable on-chain, Intent Engine that acts as a gatekeeper. Its sole purpose is to determine the outcome of an intent verification, returning only either true or false. It is only when a user’s supplied intents are immutably respected that a keychain can modify a user’s state.

Reviewer: ```suggestion
A: Warden Protocol has an immutable on-chain, Intent Engine that acts as a gatekeeper. Its sole purpose is to determine the outcome of an intent verification, returning only either true or false. It is only when a user's supplied intents are immutably respected that a keychain can modify a user's state.
```

**Q: Why are intents important?** 
A: Intents ensure that only valid transactions are processed on the blockchain, maintaining its security and integrity. They help establish trust by enforcing the rules that govern the network's operations.

**Q: How do I set my intents?** 

> A: Warden users have an Intent Configurator whic can be accessed via a GUI or over CLI. This lets a user interface and configure intents with their chosen keychain.

Reviewer: ```suggestion
A: Warden users have an Intent Configurator which can be accessed via a GUI or over CLI. This lets a user interface and configure intents with their chosen keychain.
```

***  

## SpaceWard

**Q: What is SpaceWard?** 
A: SpaceWard is an example of the powerful type of OApps that can be built on Warden. Think of SpaceWard as the front-end interface for Warden. It provides users with a user-friendly platform to access and manage their wallets, create Spaces (account aggregation environments for wallets), and interact with decentralized applications (dApps) and other components of the web3 ecosystem.

**Q: What can you do with SpaceWard?** 
A: With SpaceWard, users can securely manage accounts and wallets on any blockchain. Users can aggregate and drive all their web3 activity from SpaceWard, protected by extensible intents. Developers have an environment to build and deploy applications on both Warden and any other blockchain.

  

**Q: Why is SpaceWard important?** 
A: SpaceWard simplifies the user experience for interacting with blockchain networks and decentralized services. It provides an intuitive interface for managing wallets and accessing the features of the Warden Protocol, making blockchain technology more accessible to a wider audience.

***

## OApps  

**Q: What are OApps?** 

> A: OApps are a powerful evolution to traditional smart contracts. They consist of three parts: application & contracting logic, a stack of keychains, and a user-supplied, parameterizable intent configurator. Owing to this OApps can achieve remarkable features: they are modularly secure, omnichain interoperable and chain-abstracted.

Reviewer: ```suggestion
A: OApps are a powerful evolution of traditional smart contracts. They consist of three parts: application & contracting logic, a stack of keychains, and a user-supplied, parameterizable intent configurator. Owing to this, OApps can achieve remarkable features: they are modularly secure, omnichain interoperable, and chain-abstracted.
```

  
**Q: How does modular key management work?** 
A: With modular key management, users can select key management solutions that best suit their security and usability preferences. They can distribute the responsibility of key management across multiple keychains, enhancing security and reducing the risk of single points of failure. In the context of the Warden Protocol, modular key management ensures that users have greater control over their cryptographic keys and can choose the most suitable key management solutions for their needs.

**Q: What are the benefits of modular key management?** 
A: Modular key management provides users with flexibility, security, and interoperability in managing their cryptographic keys. It allows them to choose from a range of key management solutions, adapt to changing security requirements, and integrate with different blockchain systems and applications. By decentralizing key management, modular key management enhances the resilience and trustworthiness of blockchain networks and ensures that users maintain control over their digital assets.

 ***

## Omnichain Interoperability

**Q: How does Warden achieve omnichain interoperability?** 

> A: Warden is designed to overcome isolated and fragmented environments. The goal here is to abstract away the various boundaries between blockchains. With Warden users, and OApps, can seamlessly swap tokens across supported chains and bridge to 64 connected chains. Warden Protocol supports cross-chain transactions with Ethereum and other IBC-enabled chains, and any ECDSA or EDDSA-based chain supported by a keychain (e.g. Bitcoin).

Reviewer: ```suggestion
A: Warden is designed to overcome isolated and fragmented environments. The goal here is to abstract away the various boundaries between blockchains. With Warden, users and OApps can seamlessly swap tokens across supported chains and bridge to 64 connected chains. Warden Protocol supports cross-chain transactions with Ethereum, other IBC-enabled chains and any ECDSA or EDDSA-based chain supported by a keychain (e.g. Bitcoin).
```

***

## Chain Abstraction

**Q: How does Warden offer chain abstraction?** 

> A: Spaces are users’ gateways to the mesh network of OApps and any other blockchain. They are identity-abstracted, privacy-preserved, account-aggregated Warden addresses with which users can interact with OApps or entirely separate web3 applications. By using key ids, each OApp user can receive an infinite number of remote addresses on every ECDSA-/EDDSA based blockchain. Traditional smart contract applications only target users of a single chain. Thanks to Spaces, OApps can sign transactions and general messages targeted for any destination chain. OApps can sign transactions executed on other blockchains (e.g. writing to other chains), and thanks to a direct network integration of listeners can query data and events from other chains (e.g. reading from other chains).

Reviewer: ```suggestion
A: Spaces are users' gateways to the mesh network of OApps and any other blockchain. They are identity-abstracted, privacy-preserved, and account-aggregated Warden addresses with which users can interact with OApps or entirely separate web3 applications. By using key ids, each OApp user can receive an infinite number of remote addresses on every ECDSA-/EDDSA-based blockchain. Traditional smart contract applications only target users of a single chain. Thanks to Spaces, OApps can sign transactions and general messages targeted at any destination chain. OApps can sign transactions executed on other blockchains (e.g., writing to other chains), and thanks to a direct network integration, listeners can query data and events from other chains (e.g., reading from other chains).
```
